docs/design/autotuing-design.md

```diff
@@ -0,0 +1,269 @@
+# Design to Auto-Tuning Hyperparameters with DLRover
+
+This design doc describes how DLrover auto-tunes hyperparameters a job.
+
+## Background
+
+In the fast-paced world of deep learning, large-scale language models have
+emerged as stars in the field of natural language processing. However, training
+these heavyweight models comes with a major challenge: finding the perfect
+training settings, especially when it comes to micro-batch sizes per GPU and
+learning rates. It's all about striking the right balance between speedy
+training and accurate results. With models getting bigger and optimization
+techniques getting trickier, predicting how much GPU memory need during
+training has become  tough and uncertain for user. That's why users often end
+up trying things out repeatedly to get the right settings.
+
+## Target
+
+Our primary objective is to design and implement a performance monitoring and
+optimization module for the training process, aimed at enhancing the efficiency
+ and effectiveness of distributed deep learning training. The specific goals
+encompass:
+
+ **Collecting Training Data:** We intend to monitor and gather
+ various performance data during the training process, such as GPU memory usage
+ and utilization, along with hyperparameter configurations including batch size
+ and learning rates.
+
+ **Real-time Analysis of Training Performance Data:** Our
+ aim is to analyze the collected performance data in real-time, allowing us to
+ provide optimization strategies based on the current training status. This
+ includes suggesting new batch sizes and learning rates to maximize GPU
+ resource utilization.
+
+ **Dynamic Batch Size Configuration:** Leveraging the
+ collected and analyzed performance data, we'll devise an algorithm that
+ maximizes GPU resource utilization during task submissions while preventing
+ Out-of-Memory (OOM) errors.
```

Comment:
Here， we can emphasize that our solution does not need to restart the training process.

Reply:
Yes, now we claim our elastic dataloader as real-time configurator in no-stop manner.